Zabolotskiy/hash/main.cpp

```diff
@@ -0,0 +1,64 @@
+#include <Windows.h>
+#include <fstream>
+#include "../src/hashlibpp.h"
+using namespace std;
+
+HANDLE inputMutex, outputMutex;
+ifstream *infile;
+ofstream *outfile;
+hashwrapper *md5 = new md5wrapper();
+hashwrapper *sha1 = new sha1wrapper();
+char* block;
+int sizeBlock = 0;
+char* hashType;
+
+void Thread(void* pParams)
+{
+		block = new char[sizeBlock];
+		while (!infile->eof())
+		{
+			WaitForSingleObject(inputMutex, INFINITE);
+				infile->read((char*)(block), sizeBlock);
+			ReleaseMutex(inputMutex);
+			
+			WaitForSingleObject(outputMutex, INFINITE);
+				if (!strcmp(hashType, "md5"))	*outfile << md5->getHashFromString(block) << endl;
+				else	*outfile << sha1->getHashFromString(block) << endl;
+			ReleaseMutex(outputMutex);
+		}
+}
+
+int main ( int argc, char **argv)
```

Comment:
Checked again.
1) `char* block` is used
2) sync/async is checked, but wrongly - when I pass sync - 3 threads are created, when async - 1 thread is created
3) There are some severe problems, the hashes are not calculated correctly. I can explain today. In general - looks like there are race conditions (mutexes don't help, you need to copy data to different arrays), plus these functions will not work if null is seen in the data, need to adapt them a little or to use some different lib, plus your program produces different inputs each time when it's run, I didn't figure out why yet.